src/main/java/ui/BoardAutoCreator.java

```diff
@@ -74,6 +87,79 @@ private void createMilestoneBoard() {
                         "It is saved under the name \"" + boardName + "\".");
     }
 
+    private void createWorkAllocationBoard() {
+        logger.info("Creating " + WORK_ALLOCATION + " board");
+
+        List<TurboUser> userList = ui.logic.getRepo(ui.logic.getDefaultRepo()).getUsers();
+
+        if (userList.isEmpty()) {
+            DialogMessage.showInformationDialog("Auto-create Board - " + WORK_ALLOCATION,
+                    WORK_ALLOCATION + " board cannot be created as your" +
+                            " account has no push access to the default repo.");
+
+            return;
+        }
+
+        panelControl.closeAllPanels();
+
+        FilterPanel panelAdded;
+        int noOfPanelsToBeGenerated = Math.min(MAX_WORK_ALLOCATION_PANELS, userList.size());
+
+        List<String> filterStringsForPanels = generateListOfFilterStringsFromTurboUsers(
+                userList.subList(0, noOfPanelsToBeGenerated), "assignee:%s sort:milestone,status");
+        List<String> assigneeNamesForPanelNames = getNameListFromTurboUsers(userList.subList(0, noOfPanelsToBeGenerated));
+
+        for (int i = 0; i < noOfPanelsToBeGenerated; i++){
+            panelAdded = panelControl.addPanelAt(i);
+            panelAdded.setPanelName("Work allocated to " + assigneeNamesForPanelNames.get(i));
+            panelAdded.setFilterByString(filterStringsForPanels.get(i));
+        }
+
+        panelControl.selectFirstPanel();
+
+        String boardName = Utility.generateName(WORK_ALLOCATION,
```

Comment:
`generateName` name is not very indicative of what it does. Not sure what's a better name though.